contracts/src/Gateway.sol

```diff
@@ -136,45 +137,46 @@ contract Gateway is IGateway, IInitializable {
         // Otherwise malicious relayers can break the bridge by allowing the message handlers below to run out gas and fail silently.
         // In this scenario case, the channel's state would have been updated to accept the message (by virtue of the nonce increment), yet the actual message
         // dispatch would have failed
-        if (gasleft() < DISPATCH_GAS + BUFFER_GAS) {
+        uint256 dispatchGas = (message.dispatchGas > 0 ? message.dispatchGas : DISPATCH_GAS);
+        if (gasleft() < dispatchGas + BUFFER_GAS) {
```

Comment:
Is the `BUFFER_GAS` still nessessary here? as we've already left some margin for the substrate config.